### ISSUE TITLE
Building on Mac

### ISSUE THREAD

**Issue by marioyc** [facebook/UETorch#13]
Some notes on how to do this in case someone needs it:

1. It is necessary to change the "liblua.so" in ScriptPlugin.Build.cs to "liblua.dylib"
2. The line "ln -sfT $TORCH_BIN/.. install" fails on Mac but removing the "T" argument seems to do it as well
3. It seems like the "RenderCore" and "Slate" references are missing in UETorch.Build.cs

**Comment by marioyc** [comment]
Also, I added the Slate dependencies to PublicDependencyModuleNames, but according to the [docs](https://docs.unrealengine.com/latest/INT/Programming/Slate/InGameUI/index.html) they should be added to PrivateDependencyModuleNames.

**Comment by Buzzard0190** [comment]
Have you been able to get this to work with the version working on 4.12?  I attempted to follow your steps and currently the only steps that applied are number 2 and 3 but only adding RenderCore. I do get an error on 3 if I add in Slate.  Everything after that seems to run well until I get to the make at which point there is no make file to run.  Any ideas on what I am missing?

**Comment by adamlerer** [comment]
No, sorry, I have not tried on Mac. You'll definitely need to fix a few things as @marioyc mentioned; if I get around to it I will try to add support so you don't need to make these changes.

Have you tried following the instructions for installing on Mac (https://github.com/EpicGames/UnrealEngine#mac) instead of using the Makefile?

**Comment by eulerwang** [comment]
I have followed the instruction , but , when I run  UE4Editor , I  get the errors  :

LoginId:e76ef2c9ba43599ccbda2db1df8794f8
EpicAccountId:2a1a6f1a3cc9401fab00a4fb4f234b86

 SEGV_MAPERR at 0x0

wcslen Address = 0x7fffc7459ba2 (filename not found) [in libsystem_c.dylib]
FStandardPlatformString::GetVarArgs(wchar_t*, unsigned long, int, wchar_t const*&, __va_list_tag*) Address = 0x100197619 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Core/Private/GenericPlatform/StandardPlatformString.cpp, line 149] [in UE4Editor-Core.dylib]
FMsg::Logf_Internal(char const*, int, FName const&, ELogVerbosity::Type, wchar_t const*, ...) Address = 0x1003b64c4 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Core/Public/Misc/CString.h, line 698] [in UE4Editor-Core.dylib]
FLuaUtils::CallFunction(lua_State*, char const*) Address = 0x16807e9b5 [Runtime/Core/Public/Containers/ContainerAllocationPolicies.h, line 484] [in UE4Editor-ScriptPlugin.dylib]
FLuaContext::Initialize(FString const&, UObject*) Address = 0x1680809e0 [/Users/eulerwang/UnrealEngine/Engine/Plugins/ScriptPlugin/Source/ScriptPlugin/Private/LuaIntegration.cpp, line 625] [in UE4Editor-ScriptPlugin.dylib]
UTorchPluginComponent::OnRegister() Address = 0x16d162ba7 [/Users/eulerwang/UnrealEngine/Engine/Plugins/UETorch/Source/UETorch/Private/TorchContext.cpp, line 25] [in UE4Editor-UETorch.dylib]
UActorComponent::ExecuteRegisterEvents() Address = 0x1021cdd01 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/Components/ActorComponent.cpp, line 1145] [in UE4Editor-Engine.dylib]
UActorComponent::RegisterComponentWithWorld(UWorld*) Address = 0x1021cca81 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/Components/ActorComponent.cpp, line 878] [in UE4Editor-Engine.dylib]
AActor::IncrementalRegisterComponents(int) Address = 0x101cc8164 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/Actor.cpp, line 4003] [in UE4Editor-Engine.dylib]
ULevel::IncrementalUpdateComponents(int, bool) Address = 0x10281a04d [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/Level.cpp, line 875] [in UE4Editor-Engine.dylib]
UWorld::UpdateWorldComponents(bool, bool) Address = 0x1034afc67 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/World.cpp, line 1391] [in UE4Editor-Engine.dylib]
UWorld::InitializeActorsForPlay(FURL const&, bool) Address = 0x1034c57a5 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/World.cpp, line 3129] [in UE4Editor-Engine.dylib]
UGameInstance::StartPIEGameInstance(ULocalPlayer*, bool, bool, bool) Address = 0x1025d4717 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Engine/Private/GameInstance.cpp, line 255] [in UE4Editor-Engine.dylib]
UEditorEngine::CreatePIEGameInstance(int, bool, bool, bool, bool, float) Address = 0x108dcbb70 [/Users/eulerwang/UnrealEngine/Engine/Source/Editor/UnrealEd/Private/PlayLevel.cpp, line 3315] [in UE4Editor-UnrealEd.dylib]
UEditorEngine::PlayInEditor(UWorld*, bool) Address = 0x108dc1064 [/Users/eulerwang/UnrealEngine/Engine/Source/Editor/UnrealEd/Private/PlayLevel.cpp, line 2426] [in UE4Editor-UnrealEd.dylib]
UEditorEngine::StartQueuedPlayMapRequest() Address = 0x108da97c2 [/Users/eulerwang/UnrealEngine/Engine/Source/Editor/UnrealEd/Private/PlayLevel.cpp, line 1135] [in UE4Editor-UnrealEd.dylib]
UEditorEngine::Tick(float, bool) Address = 0x1085ff443 [/Users/eulerwang/UnrealEngine/Engine/Source/Editor/UnrealEd/Private/EditorEngine.cpp, line 1316] [in UE4Editor-UnrealEd.dylib]
UUnrealEdEngine::Tick(float, bool) Address = 0x1092c1b07 [/Users/eulerwang/UnrealEngine/Engine/Source/Editor/UnrealEd/Private/UnrealEdEngine.cpp, line 371] [in UE4Editor-UnrealEd.dylib]
FEngineLoop::Tick() Address = 0x1000215c4 [Runtime/Core/Public/Stats/Stats2.h, line 1673] [in UE4Editor]
GuardedMain(wchar_t const*) Address = 0x1000296ff [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Launch/Private/Launch.cpp, line 52] [in UE4Editor]
-[UE4AppDelegate runGameThread:] Address = 0x10003c66c [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Launch/Private/Mac/LaunchMac.cpp, line 203] [in UE4Editor]
-[FCocoaGameThread main] Address = 0x1002e1626 [/Users/eulerwang/UnrealEngine/Engine/Source/Runtime/Core/Private/Mac/CocoaThread.cpp, line 375] [in UE4Editor-Core.dylib]
Unknown() Address = 0x7fffb3685bcd (filename not found) [in Foundation]
_pthread_body Address = 0x10aae1bc8 (filename not found) [in libsystem_pthread.dylib]
_pthread_body Address = 0x10aae1b14 (filename not found) [in libsystem_pthread.dylib]
thread_start Address = 0x10aae12fd (filename not found) [in libsystem_pthread.dylib]


My system is MacOS 10.12.4 , My Xcode is 8.2.1 , and I have followed  marioyc 's instructions  .
so , May I get more help to let this great library run on my Mac?
